webviz_subsurface/_datainput/pvt_data.py

```diff
@@ -152,124 +154,181 @@ def load_pvt_dataframe(
+        gas = Gas.from_ecl_init_file(ecl_init_file, True)
+        water = Water.from_ecl_init_file(ecl_init_file, True)
 
         column_pvtnum = []
-        column_oil_gas_ratio = []
+        column_ratio = []
         column_volume_factor = []
+        column_volume_factor_unit = []
         column_pressure = []
+        column_pressure_unit = []
         column_viscosity = []
+        column_viscosity_unit = []
         column_keyword = []
 
+        ratios: List[float] = []
+        pressures: List[float] = []
+        (pressure_min, pressure_max) = (0.0, 0.0)
+
+        if oil and not oil.is_dead_oil_const_compr():
+            (pressure_min, pressure_max) = oil.range_independent(0)
+        elif gas:
+            (pressure_min, pressure_max) = gas.range_independent(0)
+        else:
+            raise NotImplementedError("Missing PVT data")
+
+        for pressure_step in range(0, 20 + 1):
+            pressures.append(
+                pressure_min + pressure_step / 20.0 * (pressure_max - pressure_min)
+            )
+            ratios.append(0.0)
+
         if oil:
-            if len(oil.tables()) > 0 and isinstance(oil.tables()[0], LiveOil):
+            if oil.is_live_oil():
                 keyword = "PVTO"
-            elif len(oil.tables()) > 0 and isinstance(oil.tables()[0], DeadOil):
+            elif oil.is_dead_oil():
                 keyword = "PVDO"
+            elif oil.is_dead_oil_const_compr():
+                keyword = "PVCDO"
             else:
                 raise NotImplementedError(
-                    f"Oil of type '{type(oil)}' is not supported yet."
+                    "The PVT property type of oil is not implemented."
                 )
 
-            for table_index, table in enumerate(oil.tables()):
-                for outer_pair in table.get_values():
-                    for inner_pair in outer_pair.y:
-                        if isinstance(inner_pair, VariateAndValues):
-                            column_pvtnum.append(table_index + 1)
-                            column_keyword.append(keyword)
-                            if isinstance(table, DeadOil):
-                                column_oil_gas_ratio.append(0.0)
-                            else:
-                                column_oil_gas_ratio.append(outer_pair.x)
-                            column_pressure.append(inner_pair.x)
-                            column_volume_factor.append(
-                                1.0 / inner_pair.get_values_as_floats()[0]
-                            )
-                            column_viscosity.append(
-                                inner_pair.get_values_as_floats()[0]
-                                / inner_pair.get_values_as_floats()[1]
-                            )
+            for region_index, region in enumerate(oil.regions()):
+                if oil.is_dead_oil_const_compr():
+                    column_pvtnum.extend([region_index for _ in pressures])
```

Comment:
Eclipse index starts at 1 (see table_index in removed code above)
```suggestion
                    column_pvtnum.extend([region_index + 1 for _ in pressures])
```